src/core/gui/PdfFloatingToolbox.cpp

```diff
@@ -27,7 +28,7 @@
 #include "MainWindow.h"  // for MainWindow
 
 PdfFloatingToolbox::PdfFloatingToolbox(MainWindow* theMainWindow, GtkOverlay* overlay):
-        theMainWindow(theMainWindow), position({0, 0}) {
+        theMainWindow(theMainWindow), overlay(overlay), position({0, 0}) {
```

Comment:
The default GObjectSPtr constructor adopts the pointed element (so the ref count is not increased). This is targetted at GObjectSPtr p(g_new_object(...)).

In this instance, I'm assuming the GtkOverlay is not supposed to be owned by the PdfFloatingToolbox, right? So I'd change this 
```suggestion
        theMainWindow(theMainWindow), overlay(overlay, xoj::util::GObjectSPtr<GtkOverlay>::ref_sink), position({0, 0}) {
```
(seems safer than just `ref` in case the ref was still floating, but it really should not be floating at this point)